tests/docs/writing-integration-test.md

```diff
@@ -0,0 +1,53 @@
+# Writing an integration test
+
+## Introduction
+
+Integration tests are run against a live running daprd binary locally. Each test
+scenario is run against a new instance of daprd, where each scenario modifies
+the daprd configuration to suit the test. Tests are expected to complete within
+seconds, ideally less than 5, and should not take longer than 30. Daprd is
+always built from source within the test.
+
+
+## Invoking the test
+
+`go test -v --race ./tests/integration`
```

Comment:
Using `--race` is not ideal in "normal testing" because it uses CGo.

Reply:
Why is that a bad thing?

Reply:
We build and ship dapr with CGo disabled. Running tests with CGo enabled could cause some issues we haven't yet figured out.
Also tests with `--race` are slower and they require rebuilding Dapr and all deps with CGo support :)

Reply:
> could cause some issues

What issues would this lead to?

> slower and they require rebuilding Dapr

You're going to be building dapr to run these tests as is stands, and currently, this test takes the same amount of time with or without `--race`- about 2 seconds.

 ---

Regardless of all this, developers should always be testing their software with a race checker.

Reply:
I have added a commit to ensure that CGO is disabled when building the daprd binary.

https://github.com/dapr/dapr/pull/6327/commits/540a2429d85f6c588b016db4257116b63b36d49d